src/ClientHandler.java

```diff
@@ -34,34 +32,116 @@ public void run() {
 		Message message = null;
 		is_logged_in = false;
 		currUser = null;
+		
+		//TODO: remove
+		printDebug();
     
     try {
     	
     	//Generate input and output stream objects.
     	outObj = new ObjectOutputStream(clientSocket.getOutputStream());
     	inObj = new ObjectInputStream(clientSocket.getInputStream());
     	
+    	
     	//Authentication Loop
     	do {
     		//See if message came in
     		if (inObj.available() > 0) {
         		message = (Message) inObj.readObject();
         		
-        		//Authenticate user
-        		currUser = authenticator.authenticate(message.getUserId(), message.getPassword());
-    		}
-    		
-    		if (currUser != null) {
-    			is_logged_in = true;
+        		if (message.getType() == MessageType.Login) {
+            		//Authenticate user
+            		currUser = authenticator.authenticate(message.getUsername(), message.getPassword());
+        		}
+        		
+        		if (currUser != null) {
+        			is_logged_in = true;
+        		}
     		}
 
     	} while (!is_logged_in);
     	
+    	//Send client all of its relevant user data
+    	message = new Message(MessageType.Login);
+    	message.setContents("Login Success");
+    	message.setUserId(currUser.getId());
     	
+    	//get list of room IDs that a user is in
+    	//then get the list of rooms based on those IDs 
+    	message.setRooms(rooms.getRoomsForUser(users.getUserRooms(user_id)));
```

Comment:
`user_id` [property](https://github.com/NewtonGluten/CS401-Group-Four-Project/pull/6/files#diff-92514a3d36dbcd6cbdae747d9a7f28d9ce0e58a959956f7361788869c6d33a83L13) is never set before it's used so `getUserRooms` returns `null` which causes an error in `getRoomsForUser`